server/sonar-server/src/main/java/org/sonar/server/permission/index/AuthorizationDao.java

```diff
@@ -0,0 +1,218 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.permission.index;
+
+import com.google.common.collect.Lists;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.dbutils.DbUtils;
+import org.apache.commons.lang.StringUtils;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+
+import static org.sonar.db.DatabaseUtils.executeLargeInputs;
+import static org.sonar.db.DatabaseUtils.repeatCondition;
+
+/**
+ * No streaming because of union of joins -> no need to use ResultSetIterator
+ */
+public class AuthorizationDao {
+
+  public static final class Dto {
+    private final String projectUuid;
+    private final long updatedAt;
+    private final List<String> users = Lists.newArrayList();
+    private final List<String> groups = Lists.newArrayList();
+
+    public Dto(String projectUuid, long updatedAt) {
+      this.projectUuid = projectUuid;
+      this.updatedAt = updatedAt;
+    }
+
+    public String getProjectUuid() {
+      return projectUuid;
+    }
+
+    public long getUpdatedAt() {
+      return updatedAt;
+    }
+
+    public List<String> getUsers() {
+      return users;
+    }
+
+    public Dto addUser(String s) {
+      users.add(s);
+      return this;
+    }
+
+    public Dto addGroup(String s) {
+      groups.add(s);
+      return this;
+    }
+
+    public List<String> getGroups() {
+      return groups;
+    }
+  }
+
+  private static final String SQL_TEMPLATE = "SELECT " +
+    "  project_authorization.project as project, " +
+    "  project_authorization.login as login, " +
+    "  project_authorization.permission_group as permission_group, " +
+    "  project_authorization.updated_at as updated_at " +
+    "FROM ( " +
+
+    // project is returned when no authorization
+    "      SELECT " +
+    "      projects.uuid AS project, " +
+    "      projects.authorization_updated_at AS updated_at, " +
+    "      NULL AS login, " +
+    "      NULL  AS permission_group " +
+    "      FROM projects " +
+    "      WHERE " +
+    "        projects.qualifier = 'TRK' " +
+    "        AND projects.copy_component_uuid is NULL " +
+    "        {projectsCondition} " +
+    "      UNION " +
+
+    // users
+
+    "      SELECT " +
+    "      projects.uuid AS project, " +
+    "      projects.authorization_updated_at AS updated_at, " +
+    "      users.login  AS login, " +
+    "      NULL  AS permission_group " +
+    "      FROM projects " +
+    "      INNER JOIN user_roles ON user_roles.resource_id = projects.id AND user_roles.role = 'user' " +
+    "      INNER JOIN users ON users.id = user_roles.user_id " +
```

Comment:
as suggested in ProjectMeasuresIndexDefinition, replacing logins by user ids would allow to remove the inner join on table `users`.